specification/sql/resource-manager/Microsoft.Sql/preview/2021-11-01-preview/EndpointCertificates.json

```diff
@@ -0,0 +1,296 @@
+        "responses": {
+          "200": {
+            "description": "Successfully retrieved a list of endpoint certificates.",
+            "schema": {
+              "$ref": "#/definitions/EndpointCertificateListResult"
+            }
+          },
+          "default": {
+            "description": "*** Error Responses: ***\n\n * 404 SubscriptionDoesNotHaveServer - The requested server was not found\n\n * 404 ServerNotInSubscriptionResourceGroup - Specified server does not exist in the specified resource group and subscription."
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "Get a list of endpoint certificates.": {
+            "$ref": "./examples/EndpointCertificatesListByInstance.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "EndpointCertificateProperties": {
+      "description": "The properties of an endpoint certificate.",
+      "type": "object",
+      "properties": {
+        "publicBlob": {
+          "description": "The certificate public blob",
+          "type": "string"
+        }
+      }
+    },
+    "Resource": {
+      "description": "ARM resource.",
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "Resource ID.",
+          "type": "string",
+          "readOnly": true
+        },
+        "name": {
+          "description": "Resource name.",
+          "type": "string",
+          "readOnly": true
+        },
+        "type": {
+          "description": "Resource type.",
+          "type": "string",
+          "readOnly": true
+        }
+      },
+      "x-ms-azure-resource": true
+    },
+    "ProxyResource": {
+      "description": "ARM proxy resource.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ],
+      "properties": {}
+    },
+    "EndpointCertificate": {
+      "description": "Certificate used on an endpoint on the Managed Instance.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/ProxyResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/EndpointCertificateProperties",
+          "description": "Resource properties.",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "EndpointCertificateListResult": {
+      "description": "A list of endpoint certificates on the target instance.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "Array of results.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/EndpointCertificate"
+          },
+          "readOnly": true
+        },
+        "nextLink": {
+          "description": "Link to retrieve next page of results.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    }
+  },
+  "parameters": {
```

Comment:
>parameters [](http://example.com/codeflow?start=3&length=10)

Most of these are unused and should be removed. Of the ones that are used, SubscriptionIdParameter, ApiVersionParameter, ResourceGroupParameter have counterparts in common types. Recommend using those instead. If you do that and remove the unused ones, the only parameter you need here is ManagedInstanceNameParameter.